lib/topologies/read_preference.js

```diff
@@ -54,13 +55,54 @@ var ReadPreference = function(preference, tags, options) {
   }
 };
 
+// Support the deprecated `preference` property introduced in the porcelain layer
+Object.defineProperty(ReadPreference.prototype, 'preference', {
```

Comment:
In the future we should replace this whole thing with a class and a getter here

Reply:
agreed, but a bit pressed for time today :)